cmd/csi-node-driver-registrar/main.go

```diff
@@ -66,9 +67,10 @@ var (
 	pluginRegistrationPath  = flag.String("plugin-registration-path", "/registration", "Path to Kubernetes plugin registration directory.")
 	kubeletRegistrationPath = flag.String("kubelet-registration-path", "", "Path of the CSI driver socket on the Kubernetes host machine.")
 	healthzPort             = flag.Int("health-port", 0, "(deprecated) TCP port for healthz requests. Set to 0 to disable the healthz server. Only one of `--health-port` and `--http-endpoint` can be set.")
-	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including the health check indicating whether the registration socket exists, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--health-port` and `--http-endpoint` can be set.")
+	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including pprof and the health check indicating whether the registration socket exists, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--health-port` and `--http-endpoint` can be set.")
 	showVersion             = flag.Bool("version", false, "Show version.")
 	mode                    = flag.String("mode", ModeRegistration, `The running mode of node-driver-registrar. "registration" runs node-driver-registrar as a long running process. "kubelet-registration-probe" runs as a health check and returns a status code of 0 if the driver was registered successfully, in the probe definition make sure that the value of --kubelet-registration-path is the same as in the container.`)
+	enableProfile           = flag.Bool("enable-pprof", false, "enable pprof profiling")
```

Comment:
Just for my curiosity, can you elaborate what issue you are seeing where you want to enable pprof? I would have expected that other sidecars like provisioner or attacher would experience much more load than node-driver-registrar.

Reply:
In the latest node-driver-registrar release [v2.3.0](https://github.com/kubernetes-csi/node-driver-registrar/releases/tag/v2.3.0) with the addition of liveness probe check, I'm seeing the container consume more memory in case of linux and windows. This is a considerable increase compared to v2.2.0 release. Adding this flag to make it easier for debugging this scenario. Also, we should maybe add this to the other sidecars as well?

Reply:
Yes I think this could be useful to enable for other sidecars too. Can we quickly double check that the other sidecars would be able to use the same cli option? I would like to make sure we try to keep args consistent with other sidecars as much as possible.

Reply:
Yeah the other sidecars that have `http-endpoint` also use a `ServeMux`. We could register the handler the same way and gate it by an `enable-pprof` switch.

e.g. https://github.com/kubernetes-csi/external-provisioner/blob/7ae61466a611061ba70da2679358c8bfc8546b6f/cmd/csi-provisioner/csi-provisioner.go#L496

Reply:
Let me know if we want to add this option in all the CSI sidecars. I would be happy to open a PR.

Reply:
Yes I think that would be valuable, thanks!